pkg/k8s/client.go

```diff
@@ -184,3 +184,24 @@ func (c *Client) SetUpEventHandler(ctx context.Context) record.EventRecorder {
 	log.V(1).Info("Successfully added event broadcaster")
 	return eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "iam-manager"})
 }
+
+//GetServiceAccount returns the service account with a given name in a given namespace
+func (c *Client) GetServiceAccount(ctx context.Context, ns string, name string) *v1.ServiceAccount {
+	log := log.Logger(ctx, "k8s", "client", "GetServiceAccount")
+	log.WithValues("namespace", ns)
+	log.Info("Retrieving service account")
+	sa := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
Do you need to set this metadata here? Isn't this automatically parsed in rCl.Get call?